### ISSUE TITLE
Ampq package dependency is set to master, leading to brittle releases

### ISSUE THREAD

**Issue by serbrech** [Azure/azure-event-hubs-go#35]
### Expected Behavior
The title explains the root issue, the consequence is that this commit : https://github.com/vcabbage/amqp/commit/390d7eaeac0368026b93a13f4f7bc30d903c61cd added an error parameter to msg.Reject()

When building the project, or a project using the lib :
```bash
vendor\github.com\Azure\azure-event-hubs-go\receiver.go:207:13: not enough arguments in call to msg.Reject
        have ()
        want (*amqp.Error)
```

### Resolution
Fix the version of the dependencies to avoid breaking changes sneaking in

### Environment
- OS: any
- Go version: any
- Version of Library: latest

**Comment by serbrech** [comment]
workaround for `azure-event-hubs-go` consumers :

add 

```toml
  [[override]]
  name = "pack.ag/amqp"
  revision = "970864c81c273f6ce2faa6f1d9ed8a5451a0d63c"
```  

to your Gopkg.toml that will fix the version of the amqp package to the commit just before the api change.

**Comment by serbrech** [comment]
Oh Thanks! I'll send a PR to fix the version to 4.2, but that needs to be done in the http://github.com/azure/azure-amqp-common-go as well or be overriden in the gopkg.

**Comment by devigned** [comment]
I generally would not have used a ref to master. I think there were a change that was needed from master amqp.

I'll update both libs and tag a new release today.

**Comment by devigned** [comment]
@serbrech should be fixed in v0.3.0. Please reopen if you are still running into issues.
